src/Vault.sol

```diff
@@ -495,18 +520,22 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, IClaimable, Ownable
 
   /* ============ Deposit Functions ============ */
 
-  /// @inheritdoc IERC4626
-  function deposit(
-    uint256 _assets,
-    address _receiver
-  ) external virtual override onlyVaultCollateralized returns (uint256) {
-    return _depositAssets(_assets, msg.sender, _receiver);
+  /**
+   * @inheritdoc IERC4626
+   * @dev Will revert if the Vault is under-collateralized.
+   */
+  function deposit(uint256 _assets, address _receiver) external virtual override returns (uint256) {
+    uint256 _depositedAssets = _totalSupply();
+    uint256 _withdrawableAssets = _totalAssets();
+
+    return _depositAssets(_assets, msg.sender, _receiver, _depositedAssets, _withdrawableAssets);
```

Comment:
can inline

Reply:
Fixed in this commit: https://github.com/GenerationSoftware/pt-v5-vault/pull/51/commits/029a830aec2a6519a7c7375dec0ccde99d0e9ef5